streamlit/streamlit_app.py

```diff
@@ -0,0 +1,289 @@
+        st.stop()
+    st.success('Dataset inputted.')
+    data = pd.read_csv(uploaded_file)
+    return data
+
+on = st.toggle('Use sample data from Github')
+url_data = (r'https://raw.githubusercontent.com/transferwise/wise-pizza/main/data/synth_data.csv')
+
+# select the datasource, either local or from github
+if on:
+    st.write(f'Downloading data from {url_data}!')
+    df = pd.read_csv(url_data)
+else:
+    df=load_data_upload()
+
+# show dataset preview
+st.text('Table preview')
+st.table(df.head(10))
+
+# ask the user via streamlit for the target column
+totals = st.selectbox(
+   "What is the target column that you want to analyse? e.g. GMV/revenue",
+   df.columns,
+   index=None,
+   placeholder="Select target column",
+)
+st.write('You selected:', totals)
+
+# ask the user via streamlit for the size column
+size = st.selectbox(
+   "What is the volume column of your dataset? e.g. number of users/transactions",
+   df.columns,
+   index=None,
+   placeholder="Select volume column",
+)
+st.write('You selected:', size)
+
+# ask the user via streamlit for the flag column
+flag_column = st.selectbox(
+   "What is the flag column of your dataset you wanan split it by? Ensure this column is binary",
+   df.columns,
+   index=None,
+   placeholder="Select time column",
+)
+st.write('You selected:', flag_column)
+
+# wait until flag column is added
+if not flag_column:
+        st.stop()
+
+# calculate unique flags in the dataset
+flags = sorted(df[flag_column].unique())  
+
+# show an error message if the specified flag column is not binary
+if len(flags)>2:
+    st.error('Your flag is not binary', icon="🚨")
+
+# allow users to define what's "old" and "new" in the comparison
+flags_option = st.selectbox(
+    'Which one in your data belongs to group A?',
+    (flags))
+
+# listing all potential dimensions in the dataframe. 
+# all of them are potential columns to exclude
+candidates_excluded_columns = [element for element in df.columns if element not in [totals,size,flag_column]]
+
+# list with specified columns to exclude.
+excluded_columns = st.multiselect(
+    'Please select all columns that you want to exclude from the analysis',
+    candidates_excluded_columns)
+
+# all non dimensional columns are the ones picked by the user (if any), plus the ones that indicate totals, size and flag
+non_dimensional_columns = excluded_columns + [totals,size,flag_column]
+
+# calculating dimensions
+dims = [element for element in df.columns if element not in non_dimensional_columns]
+
+# creating the dataframes for the comparison calculations
+data = df[df[flag_column] != flags_option]  # take the group to compare to
+pre_data = df[df[flag_column] == flags_option]  # take the group to be compared
+
+st.subheader('Finding the juiciest slices', divider='rainbow')
+st.text('This section does not compare groups, but rather checks which features have the most impact in the target column you selected.')
+
+## finding juiciest slices
```

Comment:
Juiciest slices are only calculated / shown when a column to split by is selected. It would make sense to have this plot shown independently of whether someone wants to investigate the changes / split by binary variables.